source/FreeRTOS_UDP_IP.c

```diff
@@ -96,28 +96,31 @@ void vProcessGeneratedUDPPacket( NetworkBufferDescriptor_t * const pxNetworkBuff
 {
     const UDPPacket_t * pxUDPPacket;
 
-    /* Map the UDP packet onto the start of the frame. */
-
-    /* MISRA Ref 11.3.1 [Misaligned access] */
-/* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
-    /* coverity[misra_c_2012_rule_11_3_violation] */
-    pxUDPPacket = ( ( UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
-
-    switch( pxUDPPacket->xEthernetHeader.usFrameType )
+    if( pxNetworkBuffer != NULL )
     {
-        #if ( ipconfigUSE_IPv4 != 0 )
-            case ipIPv4_FRAME_TYPE:
-                vProcessGeneratedUDPPacket_IPv4( pxNetworkBuffer );
+        /* Map the UDP packet onto the start of the frame. */
+
+        /* MISRA Ref 11.3.1 [Misaligned access] */
+        /* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-113 */
+        /* coverity[misra_c_2012_rule_11_3_violation] */
+        pxUDPPacket = ( ( UDPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
+
+        switch( pxUDPPacket->xEthernetHeader.usFrameType )
+        {
+            #if ( ipconfigUSE_IPv4 != 0 )
```

Comment:
Do we need v4 and v6 flags in unit tests. I think we decided that both flags will always be enabled for unit test.

Reply:
This is original source code. We need that flag to differentiate IPv4/IPv6.